internal/distro/fedora/partition_tables.go

```diff
@@ -162,6 +162,14 @@ var iotBasePartitionTables = distro.BasePartitionTableMap{
 				Size:     501 * common.MebiByte, // 501 MiB
 				Type:     "06",
 				Bootable: true,
+				Payload: &disk.Filesystem{
+					Type:         "vfat",
+					Mountpoint:   "/boot/efi",
+					Label:        "EFI-SYSTEM",
+					FSTabOptions: "umask=0077,shortname=winnt",
+					FSTabFreq:    0,
+					FSTabPassNo:  2,
+				},
```

Comment:
build fails with:
```
mkfs.fat 4.2 (2021-01-31)" jobId=cf4430ef-4bb3-4a3e-9bd7-e212999bb62f
d-worker[23450]: time="2023-04-11T04:55:21-04:00" level=info msg="    Volume ID does not fit in 32 bit" jobId=cf4430ef-4bb3-4a3e-9bd7-e212999bb62f
```

Reply:
```
        {
          "type": "org.osbuild.mkfs.fat",
          "options": {
            "volid": "ff498964a9a24500a9e297f5d95a59a2"
          },
          "devices": {
            "device": {
              "type": "org.osbuild.loopback",
              "options": {
                "filename": "disk.img",
                "start": 2048,
                "size": 1026048,
                "lock": true
              }
            }
          }
        },
```

Reply:
We use the following code to generate the volume identifier for a vfat filesystem:
```
        case "vfat":
            options := &MkfsFATStageOptions{
                VolID: strings.Replace(fsSpec.UUID, "-", "", -1),
            }
            stage = NewMkfsFATStage(options, stageDevices)
```
Which will indeed not fit into 32-bits. I'll look into fixing both at the same time.

Reply:
Right so this comes from the fact that I should've put a UUID in the vfat filesystem (which becomes the volume identifier), if you don't set that explicitly a normal UUID is generated which is too large to fit in a FAT volume id.

Reply:
perfect, I'll try the new code, thanks @supakeen